src/main/scala/com/gradle/internal/CustomBuildScanEnhancements.scala

```diff
@@ -1,328 +1,369 @@
+        azureServerUrl <- azureServerUrl
+        azureProject <- env.envVariable[String]("SYSTEM_TEAMPROJECT")
+        buildId <- buildId
+      } yield sbt.url(s"$azureServerUrl$azureProject/_build/results?buildId=$buildId")
+      val buildLink = buildUrl.map(("Azure Pipelines build", _)).orElse(azureServerUrl.map(("Azure Pipelines", _)))
+
+      val ops = Seq(
+        (bs: BuildScan) => bs.value("CI provider", "Azure Pipelines"),
+        ifDefined(buildLink) { case (bs, (label, url)) => bs.link(label, url) },
+        ifDefined(buildId)(_.value("CI build number", _))
+      )
+      Function.chain(ops)
     }
+  }
 
-    val gitRepo = execAndGetStdOut("git", "config", "--get", "remote.origin.url")
-    val gitCommitId = execAndGetStdOut("git", "rev-parse", "--verify", "HEAD")
-    val gitCommitShortId = execAndGetStdOut("git", "rev-parse", "--short=8", "--verify", "HEAD")
-    val gitBranchName = getGitBranchName()
-    val gitStatus = execAndGetStdOut("git", "status", "--porcelain")
-
-    if (isNotEmpty(gitRepo)) {
-      buildScan.addValue("Git repository", redactUserInfo(gitRepo.get))
-    }
-    if (isNotEmpty(gitCommitId)) {
-      buildScan.addValue("Git commit id", gitCommitId.get)
-    }
-    if (isNotEmpty(gitCommitShortId)) {
-      // Ensure server URL is configured by deferring call at execution time
-      addCustomValueAndSearchLink("Git commit id", "Git commit id short", gitCommitShortId.get)
-    }
-    if (isNotEmpty(gitBranchName)) {
-      buildScan.tag(gitBranchName.get)
-      buildScan.addValue("Git branch", gitBranchName.get)
-    }
-    if (isNotEmpty(gitStatus)) {
-      buildScan.tag("Dirty")
-      buildScan.addValue("Git status", gitStatus.get)
+  private val captureBuildkite: BuildScan => BuildScan = {
+    if (!CiUtils.isBuildkite) identity
+    else {
+      val prSource = for {
+        repository <- env.envVariable[String]("BUILDKITE_PULL_REQUEST_REPO")
+        webRepoUri <- Utils.toWebRepoUri(repository)
+        prNumber <- env.envVariable[String]("BUILDKITE_PULL_REQUEST")
+      } yield sbt.url(s"$webRepoUri/pull/$prNumber")
+
+      val ops = Seq(
+        (bs: BuildScan) => bs.value("CI provider", "Buildkite"),
+        ifDefined(env.envVariable[URL]("BUILDKITE_BUILD_URL"))(_.link("Buildkite build", _)),
+        ifDefined(env.envVariable[String]("BUILDKITE_COMMAND"))(withCustomValueAndSearchLink(_, "CI command", _)),
+        ifDefined(env.envVariable[String]("BUILDKITE_BUILD_ID"))(_.value("CI build number", _)),
+        ifDefined(prSource)(_.link("PR source", _))
+      )
+      Function.chain(ops)
     }
+  }
 
-    val gitHubUrl = envVariable("GITHUB_SERVER_URL")
-    val gitRepository = envVariable("GITHUB_REPOSITORY")
-    if (gitHubUrl.isDefined && gitRepository.isDefined && isNotEmpty(gitCommitId)) {
-      buildScan.link("GitHub source", gitHubUrl.get + "/" + gitRepository.get + "/tree/" + gitCommitId.get)
-    }
+  private val captureCiMetadata: BuildScan => BuildScan = {
```

Comment:
can we keep the ordering of this function to make the diff easier and make this more consistent with Java?

Reply:
I think the order is the name: Jenkins - TeamCity - Circle CI - Bamboo - GitHub Actions - GitLab - Travis - BitRise - GoCD - Azure Pipelines - Buildkite.

Or did you mean something else?

Reply:
The `captureCiMetadata` function is listed after the `captureCiOrLocal` before your changes. can we keep that ordering? This is to make it easier to compare this with the java code and maintain parity.